File: CS/TableQueryGetSql/XtraReport1.cs
#region #Reference
using System;
using DevExpress.DataAccess.ConnectionParameters;
using DevExpress.DataAccess.Sql;
using DevExpress.XtraReports.UI;
// ...
#endregion #Reference

namespace TableQueryGetSql {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        #region #Code
        private void XtraReport1_DataSourceDemanded(object sender, EventArgs e) {
            // Create a new SQL data source and specify its connection parameters.
            var sqlDataSource = new SqlDataSource(new Access97ConnectionParameters() {
                FileName = @"..//..//nwind.mdb"
            });

            // Create a query and assign it to the data source.
            TableQuery tableQuery = new TableQuery("Categories");
            TableInfo table = new TableInfo("Categories");
            table.SelectColumns("CategoryID", "CategoryName");
            tableQuery.Tables.Add(table);
            sqlDataSource.Queries.Add(tableQuery);

            //  Open the connection to the data source.
            sqlDataSource.Connection.Open();
            // Obtain the database schema of the connected data source.
            var sql = tableQuery.GetSql(sqlDataSource.Connection.GetDBSchema());

            // Assign the data source to the report.
            this.DataSource = sqlDataSource;
            this.DataMember = "Categories";
        }
        #endregion #Code


        private void Detail_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            // Display the obtained data in the report.
            this.xrLabel1.DataBindings.AddRange(new XRBinding[] {
            new DevExpress.XtraReports.UI.XRBinding("Text", null, "Categories.CategoryName")});
        }

    }
}

File: VB/TableQueryGetSql/XtraReport1.vb
#Region "#Reference"
Imports System
Imports DevExpress.DataAccess.ConnectionParameters
Imports DevExpress.DataAccess.Sql
Imports DevExpress.XtraReports.UI
' ...
#End Region ' #Reference

Namespace TableQueryGetSql
    Partial Public Class XtraReport1
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        #Region "#Code"
        Private Sub XtraReport1_DataSourceDemanded(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.DataSourceDemanded
            ' Create a new SQL data source and specify its connection parameters.
            Dim sqlDataSource = New SqlDataSource(New Access97ConnectionParameters() With {.FileName = "..//..//nwind.mdb"})

            ' Create a query and assign it to the data source.
            Dim query As SelectQuery = SelectQueryFluentBuilder.AddTable("Categories").SelectColumns("CategoryID", "CategoryName").Build("MyQuery")
            sqlDataSource.Queries.Add(query)

            ' Open the connection to the data source.
            sqlDataSource.Connection.Open()

            ' Obtain the database schema of the connected data source.
            Dim sql = query.GetSql(sqlDataSource.Connection.GetDBSchema())

            ' Assign the data source to the report.
            Me.DataSource = sqlDataSource
            Me.DataMember = "MyQuery"
        End Sub
        #End Region ' #Code


        Private Sub Detail_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles Detail.BeforePrint
            ' Display the obtained data in the report.
            Me.xrLabel1.DataBindings.AddRange(New XRBinding() { New DevExpress.XtraReports.UI.XRBinding("Text", Nothing, "MyQuery.CategoryName")})
        End Sub

    End Class
End Namespace
